apps/emqx_plugins/src/emqx_plugins.erl

```diff
@@ -1372,12 +1390,14 @@ plugin_dir(NameVsn) ->
 
 -spec plugin_priv_dir(name_vsn()) -> string().
 plugin_priv_dir(NameVsn) ->
-    case read_plugin_info(NameVsn, #{fill_readme => false}) of
-        {ok, #{<<"name">> := Name, <<"metadata_vsn">> := Vsn}} ->
-            AppDir = make_name_vsn_string(Name, Vsn),
-            wrap_to_list(filename:join([plugin_dir(NameVsn), AppDir, "priv"]));
-        _ ->
-            wrap_to_list(filename:join([install_dir(), NameVsn, "priv"]))
+    maybe
+        {ok, #{<<"name">> := Name, <<"rel_apps">> := Apps}} ?=
+            read_plugin_info(NameVsn, #{fill_readme => false}),
+        {ok, AppDir} ?= app_dir(Name, Apps),
+        wrap_to_list(filename:join([plugin_dir(NameVsn), AppDir, "priv"]))
+    else
+        %% Otherwise assume the priv directory is under the plugin root directory
```

Comment:
I am not sure why this was added in the first place.
doesn't `priv` dir always reside in`.../app_name-x.y.z/` ?

Reply:
We assume the `rel_apps` included an application named same with release.
But if the application doesn't, plugin author should place an priv dir into project root.

See details below:
`rel_apps` does not have an application `otp_ts_ui` included.

<details>

```
$ bat otp_ts_ui-1.1.0/release.json
───────┬────────────────────────────────────────────────────────────────────────────────────────────────
       │ File: otp_ts_ui-1.1.0/release.json
───────┼────────────────────────────────────────────────────────────────────────────────────────────────
   1   │ {
   2   │     "name": "otp_ts_ui",
   3   │     "description": "Another amazing EMQX plugin.",
   4   │     "authors": [
   5   │         "JimMoen"
   6   │     ],
   7   │     "builder": {
   8   │         "name": "JimMoen",
   9   │         "contact": "LnJimMoen@outlook.com",
  10   │         "website": "http://example.com"
  11   │     },
  12   │     "repo": "https://github.com/emqx/emqx-plugin-template",
  13   │     "functionality": [
  14   │         "Demo"
  15   │     ],
  16   │     "compatibility": {
  17   │         "emqx": "~> 5.7"
  18   │     },
  19   │     "git_ref": "58e7c43a8e41a9fb62c5d12082b351f087b308f7",
  20   │     "built_on_otp_release": "26",
  21   │     "git_commit_or_build_date": "2024-05-29",
  22   │     "metadata_vsn": "0.1.0",
  23   │     "rel_apps": [
  24   │         "test_app-0.2.0",
  25   │         "map_sets-1.1.0"
  26   │     ],
  27   │     "rel_vsn": "1.1.0",
  28   │     "with_config_schema": false
  29   │ }
───────┴────────────────────────────────────────────────────────────────────────────────────────────────

$ tree                            
.
├── otp_ts_ui-1.1.0
│   ├── map_sets-1.1.0
│   │   └── ebin
│   │       ├── map_sets.app
│   │       └── map_sets.beam
│   ├── README.md
│   ├── release.json
│   └── test_app-0.2.0
│       └── ebin
│           ├── otp_ts_ui_app.beam
│           ├── otp_ts_ui.beam
│           ├── otp_ts_ui_cli.beam
│           ├── otp_ts_ui_sup.beam
│           └── test_app.app
├── otp_ts_ui-1.1.0.sha256
└── otp_ts_ui-1.1.0.tar.gz

6 directories, 11 files
```

</details>